src/ansi-c/c_typecast.cpp

```diff
@@ -456,6 +456,23 @@ void c_typecastt::implicit_typecast(
   implicit_typecast_followed(expr, src_type, type_qual, dest_type);
 }
 
+static bool is_array_element_alias(const namespacet& ns, const symbolt* const orig_symbol, const typet &src_type, const typet &dest_type)
```

Comment:
Given the type comparison, you're only looking for strict aliases, right? If so suggest renaming function to reflect that.